lib/queue.js

```diff
@@ -518,14 +531,40 @@ function pauseResumeGlobal(queue, pause){
   return queue.client.eval(script, keys.length, keys[0], keys[1], keys[2], keys[3], pause ? 'paused' : 'resumed');
 }
 
-Queue.prototype.run = function(concurrency){
+Queue.prototype.run = function(options){
   var promises = [];
   var _this = this;
+  var concurrency = options.concurrency;
+  var rateLimiter;
+
+  if (!options.rateLimit) {
+    rateLimiter = undefined;
+  } else {
+    var limiter = Limiter({
```

Comment:
in order to keep the run function small it would be great to refactor the rateLimiter outside of it.